.gitlab-ci.yml

```diff
@@ -0,0 +1,219 @@
+  artifacts:
+    untracked: true
+    paths:
+      - deviceauth-test
+
+test:prepare_acceptance:
+  image: docker
+  before_script: []
+  stage: test_prep
+  dependencies:
+    - test:build_acceptance
+  script:
+    - docker build -f Dockerfile.acceptance-testing -t $DOCKER_REPOSITORY:prtest .;
+    - docker save $DOCKER_REPOSITORY:prtest > tests_image.tar
+    - docker build -t $DOCKER_REPOSITORY:pr .
+    - docker run --rm --entrypoint "/bin/sh" -v $(pwd):/binary $DOCKER_REPOSITORY:pr -c "cp /usr/bin/deviceauth /binary"
+    - docker build -t testing -f tests/Dockerfile tests
+    - docker save testing > acceptance_testing_image.tar
+  cache:
+    key: "$CI_COMMIT_REF_SLUG"
+    paths:
+      - /go/src/github.com/mendersoftware/deviceauth/deviceauth-test
+    policy: pull
+  artifacts:
+    expire_in: 30min
+    paths:
+      - tests_image.tar
+      - acceptance_testing_image.tar
+      - deviceauth
+
+test:acceptance_tests:
+  image: tiangolo/docker-with-compose
+  before_script: []
+  services:
+    - docker:dind
+  stage: test
+  dependencies:
+    - test:build_acceptance
+    - test:prepare_acceptance
+  script:
+    - apk add git bash
+    - git clone https://github.com/mendersoftware/integration.git
+    - mv integration/extra/travis-testing/* tests/
+    - mv docs/* tests/
+    - mv deviceauth tests/
+    - docker load -i tests_image.tar
+    - docker load -i acceptance_testing_image.tar
+    - TESTS_DIR=$(pwd)/tests $(pwd)/tests/run-test-environment acceptance $(pwd)/integration $(pwd)/tests/docker-compose.yml ;
+    - TESTS_DIR=$(pwd)/tests $(pwd)/tests/run-test-environment acceptance $(pwd)/integration $(pwd)/tests/docker-compose-tenant.yml ;
+  artifacts:
+    expire_in: 30min
+    paths:
+      - tests/coverage-acceptance.txt
+  tags:
+    - docker
+
+test:static:
+  stage: test
+  dependencies: []
+  script:
+    - curl -sL https://deb.nodesource.com/setup_11.x | bash -
+    - apt-get -qq update
+    - apt-get install -qy --allow-unauthenticated python-pip e2tools nodejs
+    - pip install pyyaml
+    - npm install -g swagger-cli
+    # Get our own Swagger verifier
+    - wget https://raw.githubusercontent.com/mendersoftware/autodocs/master/verify_docs.py
+    # Test if code was formatted with 'go fmt'
+    # Command will format code and return modified files
+    # fail if any have been modified.
+    - if [ -n "$(go fmt)" ]; then echo 'Code is not formatted with "go fmt"'; false; fi
+    # Perform static code analysys
+    - go vet `go list ./... | grep -v vendor`
+    # Fail builds when the cyclomatic complexity reaches 15 or more
+    - gocyclo -over 15 `find . -iname '*.go' | grep -v 'vendor' | grep -v '_test.go'`
+    # Verify that the Swagger docs are valid
+    - swagger-cli validate docs/*.yml
+    # Verify that the Swagger docs follow the autodeployment requirements
+    - if test "$(ls -A docs)"; then python2.7 verify_docs.py `find docs -name "*.yml"`; fi
```

Comment:
Cleaner with something like: if test -d docs; then...

We can change to Python 3 here too after my  PR in autodocs